Makefile.pre.in

```diff
@@ -2102,6 +2102,9 @@ python-config: $(srcdir)/Misc/python-config.in Misc/python-config.sh
 	@ # Substitution happens here, as the completely-expanded BINDIR
 	@ # is not available in configure
 	sed -e "s,@EXENAME@,$(BINDIR)/python$(LDVERSION)$(EXE)," < $(srcdir)/Misc/python-config.in >python-config.py
+	@ # BINDIR might be bigger than the max shebang size and lead to failures. Let us assume we have
+	@ # python in the $PATH and let env resolve it
+	sed -i -e '1s|^#!.*|#!/usr/bin/env python3|' python-config.py
```

Comment:
Assuming "a python" is on PATH isn't good enough, python-config.py uses the sysconfig and reports values for the interpreter actually used.

The script should really be interpreted by the correct interpreter, even using "#!/usr/bin/env python3.11" wouldn't be good enough because a user may have multiple installs (for example both homebrew and anaconda).